### PR TITLE
New: add conditional statement to log __callStatic

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

By adding this conditional statement to method `__callStatic` on `Give\Log\Log`, we can make sure our Logger is only persisting when `WP_DEBUG_LOG` is enabled or if type `error` or `warning` is used.

## Affects

- Updates `Give\Log\Log` class with a conditional around `__callStatic`

## Testing Instructions

Try logging with and without `WP_DEBUG_LOG` enabled

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@JasonTheAdams this is ready for review.

**Comment by ravinderk** [comment]
@jonwaldstein One log related test is failing. Can you fix it?

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
I think this is too broad. Waiting on users to enable logging will be problematic for support. Let's adjust this to avoid logging certain log levels: info and debug. Errors, warnings, and such should _always_ log. If support (or users) want to dig deeper, then they can enable logging and it will log deeper.

Would love your thoughts on this, @Benunc.

**Comment by JasonTheAdams** [comment]
@Benunc Still hoping to get your input on this when you have a moment. 😃

**Comment by Benunc** [comment]
Agreed. Always log errors. optionally log EVERYTHING.

**Comment by JasonTheAdams** [comment]
Thank you!

@jonwaldstein Can you please update this so that errors, exceptions, and warnings are logged always, but everything else is only logged if the condition is true?

**Comment by jonwaldstein** [comment]
@JasonTheAdams updated this to always log `error` & `warning`.  Exceptions I believe just use `error` under the hood.

File: src/Log/Helpers/Environment.php [hunk 1]
```diff
@@ -28,4 +28,14 @@ public static function isLogsPage()
 
         return false;
     }
+
+    /**
+     * @unreleased
+     *
+     * @return bool
+     */
+    public static function isLogEnabled()
```
Comment (JasonTheAdams) [inline src/Log/Helpers/Environment.php:37]: Last thing: let's make this clearer: `isDebugLoggingEnabled()`

**Review by JasonTheAdams — APPROVED** [review]
One very minor tweak, but otherwise good to go!

File: src/Log/Helpers/Environment.php (hunk 1 above)
Comment (jonwaldstein) [reply src/Log/Helpers/Environment.php]: 👍  91fb46d8

**Pull request merged**